### PR TITLE
History : display of rides with most of main and detailed informations.

### PR DESCRIPTION
**Pull request by RamBoFe**
C'est ma 1ere Pull Request :) Bon voilà à vous de voir si c'est ok tout ça.

### CODE REVIEW

**Comment by Nainterceptor** [comment]
Hello, please merge master into feature/history, to retrieve changes from master

**Comment by Nainterceptor** [comment]
Travis is failing, (linter) + Conflict file + Missing API pull request

File: pages/history.vue [hunk 1]
```diff
@@ -1,3 +1,257 @@
...
+          <div class="column">Heure</div>
+          <div class="column">Chauffeur</div>
+          <div class="column">Type de course</div>
+          <div class="column">Statut</div>
+          <div class="column"><a class="button is-rounded">Exporter CSV</a></div>
+        </div>
+
+        <div class="history-container-ride">
+          <template v-for="(ride,index) in rides.data" >
+            <div  v-bind:id="'history-ride-' + index" class="columns is-multiline is-mobile history-ride">
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Fix on commit https://github.com/fabnumdef/e-chauffeur_dashboard/commit/eb53671203e394ab76def3fa4a26e69347962c6e

File: pages/history.vue [hunk 2]
```diff
@@ -1,3 +1,257 @@
...
+  export default {
+    components: {
+      ecMap,
+    },
+
+    async asyncData({ $api }) {
+      // const { data, pagination } = await $api.campuses.getCampuses('id,name');
+      const endDate = DateTime.local().toISO();
+      const startDate = DateTime.local().minus({ months: 1 }).toISO();
+      const rides = await $api.rides('BSL', '*').getRides(startDate, endDate);
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Fix on commit https://github.com/fabnumdef/e-chauffeur_dashboard/commit/eb53671203e394ab76def3fa4a26e69347962c6e

File: pages/history.vue [hunk 3]
```diff
@@ -1,3 +1,257 @@
...
+                                  .toDuration()
+                                  .toFormat(format);
+      },
+
+      getFormatDate: function(date, format) {
+        return DateTime.fromISO(date).toFormat(format);
+      },
+
+      collapseRideDetails: function (id) {
+        let elHistoRideDetails = document.getElementById('history-ride-details-' + id);
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Fix on commit https://github.com/fabnumdef/e-chauffeur_dashboard/commit/eb53671203e394ab76def3fa4a26e69347962c6e

File: pages/history.vue [hunk 4]
```diff
@@ -1,3 +1,257 @@
...
+        color: $grey-light;
+      }
+    }
+
+    .columns:not(:last-child) {
+      margin-bottom: 0;
+    }
+
+    .d-none {
+      display: none !important;
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Fix on commit https://github.com/fabnumdef/e-chauffeur_dashboard/commit/eb53671203e394ab76def3fa4a26e69347962c6e

File: pages/history.vue [hunk 5]
```diff
@@ -1,3 +1,257 @@
...
+    }
+
+    .rotate90 {
+      transform: rotate(90deg);
+    }
+
+    .w-100 {
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Fix on commit https://github.com/fabnumdef/e-chauffeur_dashboard/commit/eb53671203e394ab76def3fa4a26e69347962c6e

File: pages/history.vue [hunk 6]
```diff
@@ -1,3 +1,257 @@
...
+    },
+
+    async asyncData({ $api }) {
+      // const { data, pagination } = await $api.campuses.getCampuses('id,name');
+      const endDate = DateTime.local().toISO();
+      const startDate = DateTime.local().minus({ months: 1 }).toISO();
+      const rides = await $api.rides('BSL', '*').getRides(startDate, endDate);
+
+      return {
+        rides
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Fix on commit https://github.com/fabnumdef/e-chauffeur_dashboard/commit/eb53671203e394ab76def3fa4a26e69347962c6e

File: pages/history.vue [hunk 7]
```diff
@@ -1,3 +1,257 @@
...
+            <div  v-bind:id="'history-ride-' + index" class="columns is-multiline is-mobile history-ride">
+              <div class="column" @click="collapseRideDetails(index)">
+                <span class="icon is-small" >
+                    <fa-icon :icon="['fas', 'chevron-right']" />
+                </span>
+                {{  getFormatDate(ride.start, 'D') }}</div>
+              <div class="column">{{ getFormatDate(ride.start, 'T') }} </div>
+              <div class="column">{{ ride.driver.name }}</div>
+              <div class="column">Service</div>
+              <div class="column"><span class="tag is-info is-medium">COURSE EFFECTUEE</span></div>
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Fix on commit https://github.com/fabnumdef/e-chauffeur_dashboard/commit/21b6bf39ddc67054ef1bc2c2241b7a64676896c3

File: pages/history.vue [hunk 8]
```diff
@@ -1,3 +1,257 @@
 <template>
-  <main>History</main>
+  <div id="history">
+    <div id="history-container" class="columns ">
+      <div id="history-container-filter" class="column is-one-fifth ">
+        <div class="history-title">
+          <h1 class="">Historique</h1>
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Fix on commit https://github.com/fabnumdef/e-chauffeur_dashboard/commit/eb53671203e394ab76def3fa4a26e69347962c6e

File: pages/history.vue [hunk 9]
```diff
@@ -1,3 +1,257 @@
...
+          </template>
+        </div>
+      </div>
+    </div>
+  </div>
 </template>
+
+<script>
+  import { DateTime } from "luxon";
+  import Interval from 'luxon/src/interval';
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Deref or defer  ?

File: pages/history.vue [hunk 10]
```diff
@@ -1,3 +1,257 @@
...
+            </div>
+          </template>
+        </div>
+      </div>
+    </div>
+  </div>
 </template>
+
+<script>
+  import { DateTime } from "luxon";
```
[the remark this answers is not in the document]
Comment (RamBoFe) [reply pages/history.vue]: Fix on commit https://github.com/fabnumdef/e-chauffeur_dashboard/pull/3/commits/14539be0591c8b8e23732bb26897fe9e27b748f9

**Pull request merged**